### PR TITLE
Initial setup

### PR DESCRIPTION
**Pull request by mennenia**
Added:

- [x] Typescript
- [x] Tslint
- [x] Jest
- [x] Travis file
- [x] Original decorator from [Artsy/Emission](https://github.com/artsy/emission/pull/491)
- [x] Some project settings and ignore files

cc/ @orta @alloy

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1,4 @@
+.jest
+node_modules
+build
+yarn-error.log
```
Comment (alloy) [inline .gitignore:4]: You‘ll want to add `.DS_Store` which is also being added in this PR ;)

**Comment by alloy** [comment]
👏 🍾

File: tsconfig.json [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+        "jsx": "react",
+        "target": "es6",
+        "module": "commonjs",
+        "rootDir": "src",
+        "outDir": "build",
+        "allowJs": true,
+        "pretty": true,
+        "moduleResolution": "node",
+        "strictNullChecks": false,
+        "experimentalDecorators": true
```
Comment (orta) [inline tsconfig.json]: Can you add https://github.com/danger/danger-js/blob/master/tsconfig.json#L6 - `"declaration": true,`

Then consumers get types 👍

File: .gitignore (hunk 1 above)
Comment (mennenia) [reply .gitignore:4]: dohhh. Thanks!

File: tsconfig.json (hunk 2 above)
Comment (mennenia) [reply tsconfig.json]: yup yup!

**Comment by mennenia** [comment]
Done! Mergin'

**Pull request merged**